apps/dispatch/dispatch-1.py

```diff
@@ -167,28 +167,15 @@ def _run( self, args ) :
 		tasks = []
 		# fallback to nodes for backwards compatibility
 		for taskName in args["tasks"] or args["nodes"] :
-			if args["script"].value :
-				task = script.descendant( taskName )
-				if task is None :
-					IECore.msg( IECore.Msg.Level.Error, "gaffer dispatch", "\"%s\" does not exist." % taskName )
-					return 1
-			else :
-				task = self.__create( taskName )
-				if task is None :
-					return 1
-				if args["applyUserDefaults"].value :
-					Gaffer.NodeAlgo.applyUserDefaults( task )
-				script.addChild( task )
+			task = self.__acquireNode( taskName, script, args )
+			if not task :
+				return 1
 			tasks.append( task )
 
 		nodes = []
 		for nodeName in args["nodes"] :
-			if not args["script"].value :
-				IECore.msg( IECore.Msg.Level.Error, "gaffer dispatch", "\"nodes\" cannot be used without a script." )
-				return 1
-			node = script.descendant( nodeName )
-			if node is None :
-				IECore.msg( IECore.Msg.Level.Error, "gaffer dispatch", "\"%s\" does not exist." % nodeName )
+			node = self.__acquireNode( nodeName, script, args )
```

Comment:
I don't think calling `__acquireNode()` here makes sense - we only want to show existing nodes, not create new ones (that won't be dispatched anyway). It also means that typos are reported as a confusing "Please provide the namespaced python class for..." rather than "node does not exist"...

Reply:
Addressed in da8b7d2603bbe8ee11719231da6db91c3e92779c